src/main/java/edu/ie3/datamodel/models/input/container/JointGridContainer.java

```diff
@@ -76,4 +76,59 @@ public int hashCode() {
   public String toString() {
     return "JointGridContainer{" + "gridName='" + gridName + '\'' + '}';
   }
+
+  @Override
+  public JointGridContainerCopyBuilder copy() {
+    return new JointGridContainerCopyBuilder(this);
+  }
+
+  /**
+   * A builder pattern based approach to create copies of {@link JointGridContainer} containers with
+   * altered field values. For detailed field descriptions refer to java docs of {@link
+   * JointGridContainer}
+   *
+   * @version 3.1
+   * @since 14.02.23
+   */
+  public static class JointGridContainerCopyBuilder
+      extends GridContainerCopyBuilder<JointGridContainerCopyBuilder> {
+    private SubGridTopologyGraph subGridTopologyGraph;
+
+    /**
+     * Constructor for {@link JointGridContainerCopyBuilder}
+     *
+     * @param jointGridContainer instance of {@link JointGridContainer}
+     */
+    protected JointGridContainerCopyBuilder(JointGridContainer jointGridContainer) {
+      super(jointGridContainer);
+      this.subGridTopologyGraph = jointGridContainer.getSubGridTopologyGraph();
+    }
+
+    /**
+     * Method to alter the {@link SubGridTopologyGraph}
+     *
+     * @param subGridTopologyGraph altered subGridTopologyGraph
+     * @return child instance of {@link JointGridContainerCopyBuilder}
+     */
+    public JointGridContainerCopyBuilder subGridTopologyGraph(
+        SubGridTopologyGraph subGridTopologyGraph) {
+      this.subGridTopologyGraph = subGridTopologyGraph;
+      return childInstance();
+    }
```

Comment:
We don't want to manually build or alter the graph. This one should be built by the PSDM method. So we can get rid of this method